.github/workflows/buildImage.yml

```diff
@@ -0,0 +1,44 @@
+name: build-docker-image
+
+on: 
+  push:
+    paths:
+      - 'Dockerfile'
+      - '.github/workflows/buildImage.yml'
+
+env: 
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      
+      - name: Log in to the Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v1
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: ghcr.io/GITHUB_ACTOR/project-name:latest
```

Comment:
I think `project-name` is a placeholder that comes from a tutorial. You probably want to replace this with something uniquely meaningful to this project, e.g. `learn-maintenance-testing`